src/backend/controllers/DislikeController.js

```diff
@@ -0,0 +1,120 @@
+import { Response } from 'miragejs';
+import { requiresAuth } from '../utils/authUtils';
+
+/**
+ * All the routes related to Disliked Videos are present here.
+ * These are private routes.
+ * Client needs to add "authorization" header with JWT token in it to access it.
+ * */
+
+/**
+ * This handler handles getting videos from user's Dislikes.
+ * send GET Request at /api/user/dislikes
+ * */
+export const getDislikedVideosHandler = function (schema, request) {
+  const user = requiresAuth.call(this, request);
+  try {
+    if (!user) {
+      return new Response(
+        404,
+        {},
+        {
+          errors: ['The email you entered is not Registered. Not Found error'],
+        }
+      );
+    }
+    // Returing User's Dislikes
+    return new Response(200, {}, { dislikes: user.dislikes });
+  } catch (error) {
+    return new Response(
+      500,
+      {},
+      {
+        error,
+      }
+    );
+  }
+};
+
+/**
+ * This handler handles adding videos to user's dislikes.
+ * send POST Request at /api/user/dislikes
+ * body contains {video}
+ * */
+
+export const addItemToDislikedVideos = function (schema, request) {
+  const user = requiresAuth.call(this, request);
+  if (user) {
+    const { video } = JSON.parse(request.requestBody);
+    // check if video present in db
+    const foundVideo = this.db.videos.findBy({ _id: video._id });
+    if (foundVideo) {
+      if (user.dislikes.some((item) => item._id === video._id)) {
+        return new Response(
+          409,
+          {},
+          {
+            errors: ['The video is already in your disliked videos'],
+          }
+        );
+      }
```

Comment:
add a `try catch` block to handle errors

Reply:
try-catch isn't there in Mockbee, So, I will skip it for now. Otherwise, It will require a whole refactor of all files.